src/test/scala/firrtlTests/AnnotationTests.scala

```diff
@@ -0,0 +1,57 @@
+package firrtlTests
+
+import java.io.StringWriter
+
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.junit.JUnitRunner
+
+import firrtl.{Parser,Circuit}
+import firrtl.{
+   Named,
+   ModuleName,
+   ComponentName,
+   CircuitAnnotation,
+   StringAnnotation,
+   BrittleCircuitAnnotation,
+   UnknownCAKind,
+   Compiler,
+   CompilerResult,
+   Annotation,
+   RenameMap,
+   VerilogCompiler
+}
+
+/**
+ * An example methodology for testing Firrtl annotations.
+ */
+abstract class AnnotationSpec extends FlatSpec {
+   def parse (s: String): Circuit = Parser.parse("",s.split("\n").toIterator)
+   def compiler: Compiler
+   def input: String
+   def getResult (annotation: CircuitAnnotation): CompilerResult = {
+      val writer = new StringWriter()
+      compiler.compile(parse(input), Seq(annotation), writer)
+   }
+}
```

Comment:
FirrtlFlatSpec exists so that things can extend it and get common functions for free. We could consider putting parse as a function in FirrtlRunners or something.